Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -178,25 +178,39 @@ namespace Babylon::Polyfills::Internal
 
     void XMLHttpRequest::Open(const Napi::CallbackInfo& info)
     {
-        m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
-        SetReadyState(ReadyState::Opened);
+        try
+        {
+            m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
+            SetReadyState(ReadyState::Opened);
+        }
+        catch (...)
+        {
+            // If we have a parse error, catch and rethrow to JavaScript
+            throw Napi::Error::New(info.Env(), "Could not parse URL scheme");
+        }
     }
 
     void XMLHttpRequest::Send(const Napi::CallbackInfo& info)
     {
-        m_request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [env{info.Env()}, this](arcana::expected<void, std::exception_ptr> result) {
-            if (result.has_error())
+        if (m_readyState != ReadyState::Opened)
+        {
+            Napi::Error::New(info.Env(), "XMLHttpRequest must be opened before it can be sent").ThrowAsJavaScriptException();
+            return;
+        }
+        m_request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [env{info.Env()}, this](arcana::expected<void, std::exception_ptr> result)
             {
-                throw Napi::Error::New(env, result.error());
-            }
+                if (result.has_error())
+                {
+                    Napi::Error::New(env, result.error()).ThrowAsJavaScriptException();
+                }
```

Comment:
Right now, if urllib throws an exception, this will throw an uncatchable exception on the JavaScript side and the XHR will complete (i think successfully?). Is that what we want?

```suggestion
                if (result.has_error())
                {
                    Napi::Error::New(env, result.error()).ThrowAsJavaScriptException();
                }
```